lazygit/mock-ai-tool.sh

```
@@ -0,0 +1,50 @@
+DIFF_INPUT=$(cat)
+
+# Check if input is empty
+if [ -z "$DIFF_INPUT" ]; then
+    echo "Error: No diff input provided" >&2
+    exit 1
+fi
+
+# Analyze diff to generate contextual messages
+# This is a simple heuristic-based approach for testing
+HAS_NEW_FILE=$(echo "$DIFF_INPUT" | grep -c "^+++ b/" || true)
+HAS_DELETED_FILE=$(echo "$DIFF_INPUT" | grep -c "^--- a/" || true)
```

Shellcheck (0.11.0)

[warning] HAS_DELETED_FILE appears unused. Verify use (or export if used externally).

(SC2034)